File: boxplot.R
library(dplyr)
library(ggplot2)
library(hrbrthemes)
library(readr)

# Substitua "caminho/para/seu/arquivo.csv" pelo caminho do arquivo CSV em sua máquina
# mpg <- read.csv("caminho/para/seu/arquivo.csv", header = TRUE)

# Criar o gráfico de boxplot
  mpg %>%
    ggplot(aes(x = class, y = hwy)) + 
    geom_boxplot(fill = "#69b3a2", alpha = 0.5) +  
    ggtitle("Consumo de combustível por tipo de veículo") +
    xlab("categorias") +
    ylab("milhas por galão") +
    theme(legend.position = "left") + 
    guides(fill = guide_legend(title = NULL)) +
    theme_ipsum() +
    theme(axis.text.x = element_text(size = 14),
          axis.text.y = element_text(size = 14),
          axis.title.x = element_text(size = 14),
          axis.title.y = element_text(size = 14))

File: boxplot.py
import matplotlib.pyplot as plt
import seaborn as sns

# Substitua "caminho/para/seu/arquivo.csv" pelo caminho do arquivo CSV em sua máquina
# mpg = pd.read_csv("caminho/para/seu/arquivo.csv")

# A linha 5 substitui essa, pois aqui estou utilizando dados de uma base de dados do Python
# Carrega o conjunto de dados 'mpg' fornecido pela seaborn
mpg = sns.load_dataset('mpg')

# Criar o gráfico de boxplot
plt.figure(figsize=(10, 6))
sns.boxplot(data=mpg, x='origin', y='mpg')
plt.title("Consumo de combustível por origem de veículo")
plt.xlabel("Origem")
plt.ylabel("Milhas por galão")
plt.grid(True)
plt.show()
